js-pkg/react/src/main.tsx

```diff
@@ -35,9 +35,24 @@ type UsePresenceOptions = {
   includeSelf?: boolean
 }
 
-export function usePresence<T extends Record<string, any>>(
+export function usePresenceSetter<T extends Record<string, any>>(): (presence: T) => void {
+  const awareness = useAwareness()
+
+  const setLocalPresence = useCallback(
+    (localState: any) => {
+      if (awareness) {
+        awareness.setLocalState(localState)
+      }
+    },
+    [awareness],
+  )
+
+  return setLocalPresence
+}
+
+export function useGetPresence<T extends Record<string, any>>(
```

Comment:
If we do get rid of the `usePresence` hook like I suggested above, I think we should rename this to `usePresence`. To me `useThing` means "give me `Thing`". So `useGetThing` reads as "give me a function that gets `Thing`".

Just my two cents.